exercise_perseus_render/assets/src/vue/perseus.vue

```diff
@@ -0,0 +1,235 @@
+      // components.
+      this.itemRendererFactory = this.react.createFactory(this.perseus.ItemRenderer);
+    },
+
+    destroyed() {
+      // Clean up the global namespace pollution that Perseus necessitates.
+      delete global.React;
+      delete global.ReactDOM;
+      global.$ = this.backup$;
+      delete global.i18n;
+      delete global.Exercises;
+    },
+
+    $trNameSpace: 'perseus',
+    $trs: {
+      showScratch: 'Show scratchpad',
+      notAvailable: 'Scratchpad not available',
+      loading: 'Loading',
+      check: 'Check Answer',
+      correct: 'Correct! Next Question.',
+      incorrect: 'Incorrect, try again.',
+      hint: 'Hint',
+    },
+    props: {
+      scratchpad: {
+        type: Boolean,
+        default: false,
+      },
+      initialHintsVisible: {
+        type: Number,
+        default: 0,
+      },
+      item: {
+        type: Object,
+        validator: (obj) => [
+          // A somewhat protracted validator to ensure that our item data conforms
+          // to that expected by the Perseus ItemRenderer,
+          // c.f. https://github.com/Khan/perseus/blob/master/src/item-renderer.jsx#L35
+          'calculator',
+          'chi2Table',
+          'periodicTable',
+          'tTable',
+          'zTable',
+        ].reduce(
+            // Loop through all of the above properties and ensure that if the 'answerArea'
+            // property of the item has them, then their values are set to Booleans.
+            (prev, key) => !(!prev ||
+              obj.answerArea.hasOwnProperty(key) &&
+              typeof obj.answerArea[key] !== 'boolean'), true) &&
+            // Check that the 'hints' property is an Array.
+          Array.isArray(obj.hints) &&
+          obj.hints.reduce(
+            // Check that each hint in the hints array is an object (and not null)
+            (prev, item) => item && typeof item === 'object', true) &&
+          // Check that the question property is an object (and not null)
+          obj.question && typeof obj.question === 'object',
+        required: true,
+      },
+      problemNumber: {
+        type: Number,
+        default: 1,
+      },
+    },
+    data: () => ({
+      // Is the perseus item renderer loading?
+      loading: true,
+      // Is the current answer correct?
+      correct: false,
+      // Has an answer been submitted?
+      empty: true,
+    }),
+
+    methods: {
+      renderItem() {
+        // Reset the state tracking variables.
+        this.empty = this.loading = true;
+        this.correct = false;
+        // Clean up any existing itemRenderer.
+        this.reactDOM.unmountComponentAtNode(this.$els.perseusContainer);
+
+        // Create a new one with current item data.
+        this.itemRenderer =
+        this.reactDOM.render(this.itemRendererFactory( // eslint-disable-line new-cap
+          this.itemRenderData, null), this.$els.perseusContainer);
+        this.loading = false;
```

Comment:
wouldn't javascript will just run into this.loading even if this.reactDOM.render is not finished yet due to JS's async nature? Also since reactDOM.render can pass in a option callback, might just put `this.loading = false;` in the callback.